src/gsym.rs

```diff
@@ -74,6 +77,93 @@ impl SymResolver for GsymResolver {
         None
     }
 
+    /// Finds the source code location for a given address.
+    ///
+    /// This function takes in an address and returns the file path,
+    /// line number and column of the line in the source code where
+    /// the address corresponds to. If it doesn't find any match it
+    /// returns None.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - A 64-bit unsigned integer representing the address to find the source code location for.
+    ///
+    /// # Returns
+    ///
+    /// `Some(AddressLineInfo)` if the location is found, otherwise `None`.
+    ///
+    fn find_line_info(&self, addr: u64) -> Option<AddressLineInfo> {
+        let addr = addr - self.loaded_address;
```

Comment:
Can underflow.

Reply:
Please stop making inter-function assumptions like this. Even if this was the case, tomorrow we add a new caller and it's no longer ensured. It should be guarded against here where the operation is performed.